applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_geometry_utilities.cpp

```diff
@@ -0,0 +1,259 @@
+    // each point, meaning the local_coordinate should not have an effect
+    const auto local_coordinate = array_1d<double, 3>{0.5, 0.0, 0.0};
+    Matrix     rotation_matrix =
+        GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(geometry, local_coordinate);
+
+    // clang-format off
+    Matrix expected_rotation_matrix(2, 2);
+    expected_rotation_matrix <<= 0.5,           0.5 * sqrt(3),
+                                -0.5 * sqrt(3), 0.5; // rotation of 60 degrees clockwise
+    // clang-format on
+    KRATOS_EXPECT_MATRIX_NEAR(expected_rotation_matrix, rotation_matrix, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeometryUtilities_ReturnsCorrectRotationMatrixForInclinedInterface_WithNonUnitLength,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    PointerVector<Node> nodes;
+    nodes.push_back(Kratos::make_intrusive<Node>(1, -0.5 * std::sqrt(3), 0.5, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(2, 0.5 * std::sqrt(3), -0.5, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(3, -0.5 * std::sqrt(3), 0.5, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(4, 0.5 * std::sqrt(3), -0.5, 0.0));
+    const LineInterfaceGeometry<Line2D2<Node>> geometry(1, nodes);
+
+    const auto local_coordinate = array_1d<double, 3>{0.0, 0.0, 0.0};
+    Matrix     rotation_matrix =
+        GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(geometry, local_coordinate);
+
+    // clang-format off
+    Matrix expected_rotation_matrix(2, 2);
+    expected_rotation_matrix <<= 0.5 * sqrt(3), 0.5,
+                                -0.5,           0.5 * sqrt(3); // Rotation of 30 degrees clockwise
+    // clang-format on
+    KRATOS_EXPECT_MATRIX_NEAR(expected_rotation_matrix, rotation_matrix, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeometryUtilities_ReturnsCorrectRotationMatrixForVerticalElement_WithNonUnitLength,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    PointerVector<Node> nodes;
+    nodes.push_back(Kratos::make_intrusive<Node>(1, 0.0, -7.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(2, 0.0, 10.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(3, 0.0, -7.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(4, 0.0, 10.0, 0.0));
+    const LineInterfaceGeometry<Line2D2<Node>> geometry(1, nodes);
+
+    const auto local_coordinate = array_1d<double, 3>{0.0, 0.0, 0.0};
+    Matrix     rotation_matrix =
+        GeometryUtilities::Calculate2DRotationMatrixForLineGeometry(geometry, local_coordinate);
+
+    // clang-format off
+    Matrix expected_rotation_matrix(2, 2);
+    expected_rotation_matrix <<= 0.0, -1.0,
+                                 1.0,  0.0; // Rotation of 90 degrees counterclockwise
+    // clang-format on
+    KRATOS_EXPECT_MATRIX_NEAR(expected_rotation_matrix, rotation_matrix, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeometryUtilities_UnityRotationFor3Plus3NodedInterface_WithNonUnitLength,
```

Comment:
This test case looks very similar to the first one. This one uses a horizontal 3+3 line interface geometry, whereas the first one uses a 2+2 line interface geometry. What about reflecting that in the test case name?
```suggestion
KRATOS_TEST_CASE_IN_SUITE(GeometryUtilities_RotationMatrixForHorizontal3Plus4LineInterfaceIsIdentityMatrix,
```

Reply:
typo 3Plus4 -> 3Plus3

Reply:
Done